### PR TITLE
[CN-1296] Added OLM test suite

### PR DESCRIPTION
**Pull request by cheels**
## Description

Added OLM test suite but only for EE because we are going to remove OS soon.

### CODE REVIEW

**Review by semihbkgr — APPROVED** [review]
LGTM 👍

File: .github/workflows/e2e-olm.yaml [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+      - name: Connect to the GKE cluster
+        uses: "google-github-actions/get-gke-credentials@v2"
+        with:
+          cluster_name: ${{ env.CLUSTER_NAME }}
+          project_id: ${{ env.GCP_PROJECT_ID }}
+          location: ${{ env.GKE_ZONE }}
+
+      - name: Get GH Public IP
+        id: ip
+        uses: haythem/public-ip@v1.3
```
Comment (semihbkgr) [inline .github/workflows/e2e-olm.yaml:192]: You replaced this action in https://github.com/hazelcast/hazelcast-platform-operator/pull/1186

File: .github/workflows/e2e-olm.yaml [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+    needs: [create-gke-cluster]
+    uses: ./.github/workflows/grafana-agent.yaml
+    secrets: inherit
+    with:
+      cluster_name: ${{ needs.create-gke-cluster.outputs.CLUSTER_NAME }}
+      cluster_type: gke
+      gh_run_id: ${{ github.run_id }}
+      gh_run_number: ${{ github.run_number }}
+      gh_sha: ${{ github.sha }}
+      namespaces: "test-operator-os, test-operator-ee"
```
Comment (kutluhanmetin) [inline .github/workflows/e2e-olm.yaml:86]: There shouldn't be `test-operator-os` anymore.

**Review by kutluhanmetin — APPROVED** [review]
Just a tiny comment

**Pull request merged**